File: rb/features.rb
# DemonSlayer SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module DemonSlayerFeatures
  def self.make_feature(name)
    case name
    when "base"
      DemonSlayerBaseFeature.new
    when "ratelimit"
      DemonSlayerRatelimitFeature.new
    when "retry"
      DemonSlayerRetryFeature.new
    when "test"
      DemonSlayerTestFeature.new
    when "timeout"
      DemonSlayerTimeoutFeature.new
    else
      DemonSlayerBaseFeature.new
    end
  end
end

File: py/features.py
# DemonSlayer SDK feature factory

from feature.base_feature import DemonSlayerBaseFeature
from feature.test_feature import DemonSlayerTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DemonSlayerBaseFeature(),
        "test": lambda: DemonSlayerTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
